app-release-procedures.rst

```diff
@@ -0,0 +1,234 @@
+**********************
+App release procedures
+**********************
+
+Goal: provide simple and short guide line document on code delivery procedure
+Target: Document targets end-services only
+
+NOTE: this document is a draft version made mainly for discussions.
+
+Main line: developer is responsible for code reliability and delivery
+
+Delivery procedure phases:
+
+    #. Creating Pull Request
+    #. Code Review
+    #. Deployment
+    #. Delivery Verification
+    #. Rolling Back
+
+Creating Pull Request
+=====================
+
+PR description must have all necessary information for making code review and delivery easy. While creating PR consider that it should have following parts:
+    * **Reviewers**. Assign on all development team.
+    * **Original issue**. Link original issue and all supporting information.
+    * **Downsides**. Describe solution downsides (if exists) or share any concerns about it.
+    * **Roll-out plan**. Specify all resources that will be modified. Describe step-by-step how to deploy.
+    * **Deployment risks**. Describe potential risks of deployment.
+    * **Rollback strategy**. Develop plan how to retract release and describe it step-by-step.
+    * **Deployment assistants**. Mention ops persons, so they could consider deployment risks as well.
+
+In most cases creating PR will be copy-pasting template with small customizations.
+TODO: add reference to example templates.
+
+
+[DNM] PR
+--------
+
```

Comment:
If it involves configuration changes (Heroku env, add-ons) then it needs checklist.